src/app/services/fetch-data.service.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@angular/core';
+import { Http, Headers, RequestOptions, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import { Parser } from '../shared/parser';
+
+@Injectable()
+export class FetchDataService {
+
+  constructor(private http: Http) { }
+
+  fetchData(URL: string, data: any): Observable<any> {
+    console.log('[http service] fetch');
+    let url = URL;
+    let classPath = Parser.parseClassPath(url);
+    let headers = new Headers({ 'Content-Type': 'text/plain' });
+    let options = new RequestOptions({ headers: headers });
+    console.log(url);
+
+    switch (classPath) {
+      case 'signature': {
+        let body = 'compound=' + data.compound;
+        return this.http.post(url, body, options)
+          .map(res => ({'data': res.json(), 'type': classPath}))
+          .catch(this.handleError);
+      }
+
+      case 'compounds': {
+        let body = 'query=' + data.compound;
+        return this.http.post(url, body, options)
+          .map(res => ({'data': res.json(), 'type': classPath}))
+          .catch(this.handleError);
+      }
+
+      case 'zhang': {
+        let body = 'query=' + data.signature + ', sorted=true';
+        return this.http.post(url, body, options)
+          .map(res => ({'data': res.json(), 'type': classPath}))
```

Comment:
the `type` is saved, as the `setData()` and `getData()` methods of the manipulateDataService are implemented in a generic way, so they need to know the endpoint of the API of the current transaction.